Fix `EditDialog#onAfterSave` not called on form submission

johnnyomair opened the pull request:
The `onAfterSave` callback was only called when submitting a form inside an `EditDialog` by clicking the save button, but not when submitting the form by hitting the enter key. We fix this by adding the callback to the `EditDialogFormApi` and calling it after the form has been successfully submitted.






- [x] Add changeset (if necessary)

johnnyomair commented on packages/admin/admin/src/FinalForm.tsx:
```diff
@@ -235,6 +235,7 @@ export function FinalForm<FormValues = AnyObject>(props: IProps<FormValues>) {
                     }
 
                     onAfterSubmit(values, form);
+                    editDialogFormApi?.onAfterSave?.();
```
I'm not sure if this should be called here, or in the default `onAfterSubmit` handler. The onAfterSubmit handler may be overridden, which would result in the `onAfterSave` callback not being called (only when submitting by hitting enter, click on the save button probably still works). What do you think?

nsams replied:
good here I'd say

nsams reviewed (approved):
Don't you have a story you used for developing this?

and you could even create a unittest

johnnyomair commented:
> Don't you have a story you used for developing this?

I just added a log to the docs and removed it afterwards

> and you could even create a unittest

You're right, this would have been the correct. I will try this

nsams commented:
does the dialog get closed when submitting using enter? should it?

johnnyomair commented:
> does the dialog get closed when submitting using enter? should it?

Yes (as does pressing the "Save" button). I suppose it should

johnnyomair commented:
@nsams added tests now: f8d2a3ce4b746237ad1172187e4537936480c7f5

johnnyomair commented on packages/admin/admin/src/EditDialog.test.tsx:
```diff
@@ -0,0 +1,108 @@
+                        </RouterMemoryRouter>
+                    </MuiThemeProvider>
+                </IntlProvider>,
+            );
```
I added a custom render (as suggested by the docs) here: cb3d7564b979275f81a2bd43e6ecf6d1b5c07ab5

What do you think? One minor concern: I needed to add the `paths` and `baseUrl` options to `tsconfig.json` (and not `tsconfig.test.json`) for IDE support. I doubt that this will break something (i.e. the build), but we should verify that.

thomasdax98 replied:
Fine by me. 

Since the pipeline works, the build should be okay, right? Or do you think this can affect the application build?

johnnyomair replied:
> Since the pipeline works, the build should be okay, right?

Probably

> Or do you think this can affect the application build?

I don't think so. But we'll see once applications update to the latest version
